gui/src/installer/mod.rs

```diff
@@ -164,10 +173,36 @@ impl Installer {
 pub async fn install(ctx: Context) -> Result<PathBuf, Error> {
     let mut cfg: liana::config::Config = ctx.extract_daemon_config();
     // Start Daemon to check correctness of installation
-    let daemon = liana::DaemonHandle::start_default(cfg.clone()).map_err(|e| {
-        Error::Unexpected(format!("Failed to start daemon with entered config: {}", e))
-    })?;
-    daemon.shutdown();
+    match liana::DaemonHandle::start_default(cfg.clone()) {
+        Ok(daemon) => daemon.shutdown(),
+        Err(e) => match e {
+            StartupError::Bitcoind(BitcoindError::Wallet(path, error)) => {
+                if !matches!(error, WalletError::Creating(_)) {
+                    // Clean the created wallet directory
+                    log::warn!("Error {} while managing the wallet, trying now to delete the corrupted wallet directory", error);
+                    if let Err(e) = std::fs::remove_dir_all(path.clone()) {
```

Comment:
This is not portable to Windows, which is the purpose of having this check in the first place. On Windows, the watchonly wallet path is unfortunately the name of the wallet in the bitcoind datadir, not the path to the file.

See https://github.com/wizardsardine/liana/pull/224 and
https://github.com/wizardsardine/liana/blob/0ce888091a81932781e2f7deca765fa8d3bccabe/src/lib.rs#L189-L205